bridge/opencensus/internal/otel2oc/span_context_test.go

```diff
@@ -45,12 +57,45 @@ func TestSpanContextConversion(t *testing.T) {
 				TraceOptions: octrace.TraceOptions(0),
 			},
 		},
+		{
+			description: "trace state should be propagated",
+			input: trace.NewSpanContext(trace.SpanContextConfig{
+				TraceID:    trace.TraceID([16]byte{1}),
+				SpanID:     trace.SpanID([8]byte{2}),
+				TraceState: tsOtel,
+			}),
+			expected: octrace.SpanContext{
+				TraceID:      octrace.TraceID([16]byte{1}),
+				SpanID:       octrace.SpanID([8]byte{2}),
+				TraceOptions: octrace.TraceOptions(0),
+				Tracestate:   tsOc,
+			},
+		},
 	} {
 		t.Run(tc.description, func(t *testing.T) {
 			output := SpanContext(tc.input)
-			if output != tc.expected {
-				t.Fatalf("Got %+v spancontext, expected %+v.", output, tc.expected)
+			if !equal(output, tc.expected) {
+				t.Fatalf("Got %+v spancontext, expected %+v.", toString(output.Tracestate), toString(tc.expected.Tracestate))
 			}
 		})
 	}
 }
+
+func equal(t1, t2 octrace.SpanContext) bool {
+	return t1.IsSampled() == t2.IsSampled() &&
+		t1.SpanID == t2.SpanID &&
+		t1.TraceID == t2.TraceID &&
+		t1.TraceOptions == t2.TraceOptions &&
+		toString(t1.Tracestate) == toString(t2.Tracestate)
```

Comment:
This is not a robust way of checking equality of tracestate.

Reply:
Removed this logic after above simplification.